oauth-common/src/main/java/io/strimzi/kafka/oauth/jsonpath/ParsingContext.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2017-2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.oauth.jsonpath;
+
+class ParsingContext implements Cloneable {
+
+    char[] buffer;
+    int start;
+    int current;
+
+    ParsingContext(char[] buffer) {
+        this(buffer, 0, 0);
+    }
+
+    ParsingContext(char[] buffer, int start, int current) {
+        this.buffer = buffer;
+        this.start = start;
+        this.current = current;
+    }
+
+    @Override
+    protected ParsingContext clone() throws CloneNotSupportedException {
+        ParsingContext ctx = (ParsingContext) super.clone();
+        ctx.buffer = buffer;
+        ctx.start = start;
+        ctx.current = current;
+        return ctx;
+    }
+
+    public ParsingContext copy() {
```

Comment:
And if you're not redeclaring the throws in `clone` this method probably isn't needed?

Reply:
Well, it's not used anywhere, so I'll remove it.